### PR TITLE
Feature/onyx basic infra

### PR DESCRIPTION
**Pull request by jasebell**
Basic framework for event->S3 workflow.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,27 +6,38 @@ There are a number of environment variables that are required before the job is
 
 ### AWS Credentials
 
-AWS_ACCESS_KEY_ID=[Insert your access key]
+```
+AWS_ACCESS_KEY_ID=Insert your access key]
```
Comment (acron0) [inline README.md]: Wants another `[`

File: deployment/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+FROM java:8-jre-alpine
```
Comment (acron0) [inline deployment/Dockerfile:1]: Any reason to use alpine over phusion? Do we care?

Comment (jasebell) [reply deployment/Dockerfile:1]: This is generated from lein new onyx-app template. Not sure what the knock on would be if we changed it.

File: README.md (hunk 1 above)
Comment (jasebell) [reply README.md]: Fixed.

File: src/kixi/event2s3/core.clj [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+(defn assert-job-exists [job-name]
+  (let [jobs (methods onyx.job/register-job)]
+    (when-not (contains? jobs job-name)
+      (exit 1 (error-msg (into [(str "There is no job registered under the name " job-name "\n")
+                                "Available jobs: "] (keys jobs)))))))
+
+(defn -main [& args]
+  (let [{:keys [options arguments errors summary] :as pargs} (parse-opts args (cli-options))
+        action (first args)
+        argument (clojure.edn/read-string (second args))]
```
Comment (acron0) [inline src/kixi/event2s3/core.clj:97]: Could this argument not be included in the pargs map?

File: src/kixi/event2s3/heartbeat_server.clj [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+           [bidi.bidi :refer [tag]]
+           [bidi.vhosts :refer [make-handler vhosts-model]]
+           [com.stuartsierra.component :as component]
+           [taoensso.timbre :as timbre :refer [infof]]
+           [yada.yada :as yada :refer [resource]]
+           [yada.consume :refer [save-to-file]]
+           [yada.resources.webjar-resource :refer [new-webjar-resource]]))
+
+(defn bidi-routes []
+  ["" [["/health_check" (yada/handler "Peers are alive.\n")]]])
```
Comment (acron0) [inline src/kixi/event2s3/heartbeat_server.clj]: https://www.pivotaltracker.com/story/show/138178421/comments/161581821

Please change to `/healthcheck`

File: src/kixi/event2s3/heartbeat_server.clj [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+
+(defrecord WebServer [port listener]
+  component/Lifecycle
+  (start [component]
+    (if listener
+      component
+      (let [vhosts-model
+            (vhosts-model
+             [{:scheme :http :host (format "localhost:%d" port)}
```
Comment (acron0) [inline src/kixi/event2s3/heartbeat_server.clj]: I think you'll need to make the vhost host configurable (e.g. https://github.com/MastodonC/kixi.datastore/blob/97cf0fb22a79539de2a81500391048b1f51aa0cf/resources/config.edn#L8)
We had some trouble getting Yada to work in mesos when this was set to localhost

File: test/kixi.event2s3/jobs/basic_test.clj [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+               [test-helper :refer [with-test-env]]]
+              [onyx.plugin.core-async :refer [get-core-async-channels take-segments!]]
+              kixi.event2s3.jobs.basic
+              ;; Include function definitions
+              kixi.event2s3.tasks.math
+              onyx.tasks.core-async))
+
+(def segments [{:n 1} {:n 2} {:n 3} {:n 4} {:n 5} :done])
+
+(deftest basic-test
```
Comment (acron0) [inline test/kixi.event2s3/jobs/basic_test.clj]: Does this test pass??

File: deployment/Dockerfile (hunk 2 above)
Comment (acron0) [reply deployment/Dockerfile:1]: Best leave then

File: test/kixi.event2s3/jobs/basic_test.clj (hunk 6 above)
Comment (elisehuard) [reply test/kixi.event2s3/jobs/basic_test.clj]: I think there might be something wrong with the location vs namespace

File: src/kixi/event2s3/heartbeat_server.clj (hunk 4 above)
Comment (jasebell) [reply src/kixi/event2s3/heartbeat_server.clj]: Changed

File: src/kixi/event2s3/heartbeat_server.clj (hunk 5 above)
Comment (jasebell) [reply src/kixi/event2s3/heartbeat_server.clj]: Now uses passed in vhost name from config.

File: test/kixi.event2s3/jobs/basic_test.clj (hunk 6 above)
Comment (jasebell) [reply test/kixi.event2s3/jobs/basic_test.clj]: This can be taken out.

**Comment by elisehuard** [comment]
lgtm

**Review by elisehuard — APPROVED** [review]

**Pull request merged**